### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vladgusakovskiy**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,85 @@
-export interface Figure {
 
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Colors {
```
Comment (Githi54) [inline src/figuresClasses.ts:8]: 👏👏

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,85 @@
...
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a >= this.b + this.c
+      || this.b >= this.a + this.c
+      || this.c >= this.a + this.b
+    ) {
```
Comment (Maxim-Gumeniuk) [inline src/figuresClasses.ts:33]: make a separate variable

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,85 @@
...
+      throw new Error('Error! sides a, b and c can\'t form a triangle');
+    }
+  }
+
+  getArea(): number {
+    const perimeter = (this.a + this.b + this.c) / 2;
+
+    const sqrtValue = Math.sqrt(perimeter * (perimeter - this.a)
+    * (perimeter - this.b) * (perimeter - this.c));
```
Comment (Maxim-Gumeniuk) [inline src/figuresClasses.ts:42]: this piece of code is hard to read

**Review by Maxim-Gumeniuk — APPROVED** [review]
GJ!

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,85 @@
...
+export class Triangle implements Figure {
+  public shape = Shapes.Triangle;
 
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
```
Comment (Githi54) [inline src/figuresClasses.ts:29]: You can check it by getting largest side with Math.max(a, b, c) and calculating only once:
(a + b + c - largest) <= largest. This is shorted your code :)

File: src/figuresClasses.ts [hunk 5]
```diff
@@ -29,7 +33,7 @@ export class Triangle implements Figure {
     const maxLength: number = Math.max(a, b, c);
     const isValidLength: boolean = maxLength >= a + b + c - maxLength;
 
-    if (isValidLength) {
+    if (correctvalue(this.a, this.b, this.c) || isValidLength) {
```
Comment (AndriiRut) [inline src/figuresClasses.ts:36]: better assign sides to the variables and use it
`const a = this.a`

**Review by AndriiRut — APPROVED** [review]
🚀

**Pull request closed without merging**